owledge/spec/models/note_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'rails_helper'
+
+RSpec.describe Note, type: :model do
+  @note_list_only_parents = [{
+    title: 'Sorting Algorithms',
+    description: %w{
+      In computer science, a sorting algorithm is an algorithm that puts elements of a list in a certain order.
+    }.join(" ")
+  },{
+    title: 'Graph Theory',
+    description: %w{
+      In mathematics, graph theory is the study of graphs, which are mathematical structures
+      used to model pairwise relations between objects.
+    }.join(" ")
+  }]
+  describe 'check if Only parents function is working as expected' do
+    context 'No records so count is 0' do
+      before :each do
+        Note.delete_all
+      end
+      it 'Checks the list of parents' do
+        @notes = Note.allParents
+        expect(@notes.count).to eq(0)
+      end
+    end
+    context 'Only parents so count of function is equal to count of all' do
+      before :each do
+        Note.delete_all
+        Note.create(@note_list_only_parents)
+      end
+      it 'Checks the list of parents' do
+        @notes = Note.allParents
+        @notes_all = Note.all
+        expect(@notes.count).to eq(@notes_all.count)
```

Comment:
Nit: Try comparing @notes_all to @notes, rather than their counts and see if it works. It would be a better test, but not worth wasting a bunch of time on.

Reply:
It doesn't work since the references seem to be different, I didn't try very hard to fix it though

Reply:
Ignore this message just testing